model/fv_dynamics.F90

```diff
@@ -187,8 +188,13 @@ module fv_dynamics_mod
 
   subroutine fv_dynamics(npx, npy, npz, nq_tot,  ng, bdt, consv_te, fill,             &
                         reproduce_sum, kappa, cp_air, zvir, ptop, ks, ncnst, n_split, &
-                        q_split, u, v, w, delz, hydrostatic, pt, delp, q,             &
+                        q_split, u, v, w, delz, hydrostatic,                          &
+!The following variable is for SA-3D-TKE
+                        sa3dtke_dyco,                                                 &
```

Comment:
I would suggest not passing `sa3dtke_dyco` as a separate flag to the high-level `fv_dynamics()` routine, but instead reading it from `flagstruct%` when a calculation is needed.